tests/Spec.hs

```diff
@@ -90,31 +92,25 @@ runActionWithInit integrateResult pushResult =
         tell [AIsReviewer username]
         pure $ h $ isReviewer username
 
+-- Simulates running the action. Use the provided sha as result when integration
+-- is attempted, and use the provided push result when a push is attempted.
+runActionCustom :: Maybe Sha -> PushResult -> Action a -> (a, [ActionFlat])
+runActionCustom integrateResult pushResult = runWriter . runActionWithInit integrateResult pushResult
+
 -- Simulates running the action. Pretends that integration always conflicts.
 -- Pretends that pushing is always successful.
 runAction :: Action a -> (a, [ActionFlat])
-runAction = runWriter . runActionWithInit Nothing PushOk
+runAction = runActionCustom Nothing PushOk
 
--- Handle an event and simulate its side effects, then ignore the side effects
--- and return the new state.
+-- Handle an event, then advance the state until a fixed point,
+-- and simulate its side effects.
 handleEventFlat :: Event -> ProjectState -> (ProjectState, [ActionFlat])
 handleEventFlat event state = runAction $ handleEvent testTriggerConfig event state
 
--- Handle events and simulate their side effects, then ignore the side effects
--- and return the new state.
-handleEventsFlat :: [Event] -> ProjectState -> (ProjectState, [ActionFlat])
-handleEventsFlat events state = runAction
-  $ foldlM (flip $ handleEvent testTriggerConfig) state events
-
--- Proceed with a state until a fixed point, simulate and collect the side
--- effects.
-proceedUntilFixedPointFlat
-  :: Maybe Sha
-  -> PushResult
-  -> ProjectState
-  -> (ProjectState, [ActionFlat])
-proceedUntilFixedPointFlat integrateResult pushResult state =
-  runWriter $ runActionWithInit integrateResult pushResult $ proceedUntilFixedPoint state
+-- Handle events (advancing the state until a fixed point in between) and
+-- simulate their side effects.
+handleEvents :: [Event] -> ProjectState -> Action ProjectState
+handleEvents events state = foldlM (flip $ handleEvent testTriggerConfig) state events
```

Comment:
Great idea! I was also not a big fan of `Flat` because it doesn't mean a lot (it flattens out the free monad to a list, but that is a bit arcane). I'll rename it to `Test`, that should clarify both things at once.